packages/block-library/src/post-navigation-link/index.php

```diff
@@ -116,6 +116,11 @@ function render_block_core_post_navigation_link( $attributes, $content ) {
 		$content = $get_link_function( $format, $link );
 	}
 
+	// Return empty if the adjacent post is unavailable.
+	if ( empty( $content ) ) {
+		return '';
+	}
```

Comment:
This looks right to me so far. Thank you for the changes! 🙇🏻 

There's one backwards compatibility case that @t-hamano already mentioned: without the "empty" block wrapper, the next navigation block will align left.

We might have to think about RTL languages too, so the "previous" block should align left? Not sure there, but we should test it.

Not sure what the best thing to do here - probably CSS as @t-hamano suggests.

Here's the difference I see.

### T4 in trunk

<img width="400" alt="Screenshot 2024-08-08 at 9 56 38 AM" src="https://github.com/user-attachments/assets/154884a4-409c-480b-a8bf-8a3c978dc162">



### T4 with this PR

<img width="400" alt="Screenshot 2024-08-08 at 9 56 21 AM" src="https://github.com/user-attachments/assets/bca95524-9633-49f0-b31b-d0f11a844acc">

Reply:
This code looks good to me too 👍

> without the "empty" block wrapper, the next navigation block will align left.

This needs to be addressed. By using `maring-inline-{start|end}`, we should be able to handle RTL languages ​​as well as LTR languages.

A similar approach has been attempted for the Query Pagination block, and these two PRs may be of help:

- https://github.com/WordPress/gutenberg/pull/42764
- https://github.com/WordPress/gutenberg/pull/45651

Reply:
I'm not sure that simply removing the empty element solves all the backward compatibility issues here. 

Removing the element and adding CSS may prevent visual regressions for the default situation. There might be use cases out in the wild though where changing the block's markup breaks other theme styles.

We can avoid all BC issues by only applying the styles when needed. I don't think it's a resource-intensive option and saves having to send additional CSS.

Reply:
> We can avoid all BC issues by only applying the styles when needed. I don't think it's a resource-intensive option and saves having to send additional CSS.

I agree that preserving the element would be the ideal approach. Some themes might expecting the element to be there for styling or structural purposes.

I wonder how it'd work with global styles, presets etc. Should we strip all classnames and styles from the element?

Then there's the issue of whether themes are relying on these classnames as hooks. 🤔

Reply:
> I wonder how it'd work with global styles, presets etc. Should we strip all classnames and styles from the element?

Leveraging the [Block Selectors API](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-selectors/), we should be able to use something like the `:empty` pseudo selector in combination with `:not()` to ensure the Global Styles are only applied when they should be.

Completely untested example: `.wp-block-post-navigation-link:not(:empty)`.

I've recently used a loosely similar approach to [ensuring List and List Item global styles are only applied to top-level blocks](https://github.com/WordPress/gutenberg/pull/63540) rather than nested lists.

Reply:
Good idea! Thanks for confirming.

So maybe my [first comment](https://github.com/WordPress/gutenberg/pull/64258#issuecomment-2272493384) wasn't so wild as I thought. 🤣

Reply:
That or we're both just wild. Either way, they say great minds think alike 🤔 

I completely forgot you'd already mentioned that selector sorry. I'm a bit slow sometimes.

Still, a combination of skipping serialization of the block supports, as necessary, with the custom global styles selector works for me.